applications/asset_tracker_v2/src/main.c

```diff
@@ -159,6 +160,44 @@ static void sub_state_set(enum sub_state_type new_state)
 	sub_state = new_state;
 }
 
+/* Check the return code from nRF modem library initializaton to ensure that
+ * the modem is rebooted if a modem firmware update is ready to be applied or
+ * an error condition occurred during firmware update or library initialization.
+ */
+static void handle_nrf_modem_lib_init_ret(void)
+{
+	int ret = nrf_modem_lib_get_init_ret();
+
+	/* Handle return values relating to modem firmware update */
+	switch (ret) {
+	case 0:
+		/* Initialization successful, no action required. */
+		return;
+	case MODEM_DFU_RESULT_OK:
+		LOG_INF("MODEM UPDATE OK. Will run new firmware");
+		break;
+	case MODEM_DFU_RESULT_UUID_ERROR:
+	case MODEM_DFU_RESULT_AUTH_ERROR:
+		LOG_ERR("MODEM UPDATE ERROR %d. Will run old firmware", ret);
+		break;
+	case MODEM_DFU_RESULT_HARDWARE_ERROR:
+	case MODEM_DFU_RESULT_INTERNAL_ERROR:
+		LOG_ERR("MODEM UPDATE FATAL ERROR %d. Modem failiure", ret);
+		break;
+	default:
+		/* All non-zero return codes other than DFU result codes are
+		 * considered irrecoverable and a reboot is needed.
+		 */
+		LOG_ERR("nRF modem lib initialization failed, error: %d", ret);
+		break;
+	}
+
+	LOG_WRN("Rebooting in 5 seconds");
```

Comment:
Just wondering, why is the delay needed?

Reply:
It was just a convenience so that warning on rebooting wasn't drowning in the log from rebooting. But it isn't needed, so I removed it